strawberry/experimental/pydantic/object_type.py

```diff
@@ -106,53 +111,49 @@ def wrap(cls):
         if not fields_set:
             raise MissingFieldsListError(cls)
 
-        all_model_fields: List[Tuple[str, Any, dataclasses.Field]] = [
-            (
-                name,
-                get_type_for_field(field),
-                StrawberryField(
-                    python_name=field.name,
-                    graphql_name=field.alias if field.has_alias else None,
-                    default=field.default if not field.required else UNSET,
-                    default_factory=(
-                        field.default_factory if field.default_factory else UNSET
-                    ),
-                    type_annotation=get_type_for_field(field),
-                    description=field.field_info.description,
-                ),
-            )
-            for name, field in model_fields.items()
-            if name in fields_set
-        ]
+        all_model_fields: List[DataclassCreationFields] = []
+
+        for field_name, field in model_fields.items():
+            if field_name in fields_set:
```

Comment:
shall we do something like this to remove one nesting level?

```python
f = (field_name, field for field_name, field in model_fields.items() if field_name in fields_set)
for field_name, field in f:
    # no if here
```

Reply:
cool trick to prevent identation hell